### ISSUE TITLE
The Markdown link seems to be preventing the footnotes generation.

### ISSUE THREAD

**Issue by krymtkts** [bent10/marked-extensions#23]
- Extension Name: marked-footnote
- Extension Version: v1.1.2
- Project Version:
- Device: [e.g Desktop] all of devices
- OS@version (or Browser): Windows 11 Home 23H2 22631.2715
- Node And NPM Version:

### Description

The Markdown link seems to be preventing the footnotes generation.

Specifically, `[` causes a bug.

**To Reproduce** _(⚠️ read below)_

https://replit.com/@krymtkts/marked-footnote-repro#index.html

```html
<!doctype html>
<html lang="en">
  <body class="markdown-body">
    <div id="content"></div>

    <script src="https://cdn.jsdelivr.net/npm/marked/marked.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/marked-footnote/dist/index.umd.min.js"></script>
    <script>
      const md = `# Example

[^1]: works fine.

[^2]: not work with [link](https://github.com/bent10/marked-extensions/tree/main/packages/footnote).

it works fine[^1]. not work with link[^2].
`

      document.getElementById("content").innerHTML = new marked.Marked()
        .use(markedFootnote())
        .parse(md)
    </script>
  </body>
</html>
```

### Expected Behavior

Footnotes are not generated.

### Actual Behavior

Footnotes are generated.

**Comment by bent10** [comment]
Thanks for reporting this issue!

I've released [marked-footnote@1.1.3](https://github.com/bent10/marked-extensions/releases/tag/marked-footnote%401.1.3) that fixed this issue. Feel free to explore and let me know if you have any feedback.

**Comment by krymtkts** [comment]
It works as expected. Thanks!
